integration-tests/infinispan/src/test/java/org/apache/camel/quarkus/component/infinispan/InfinispanServerTestResource.java

```diff
@@ -54,24 +55,48 @@
-                    "camel.component.infinispan.sasl-mechanism", "DIGEST-MD5",
-                    "camel.component.infinispan.security-server-name", "infinispan");
+            if (SystemUtils.IS_OS_LINUX) {
+                return CollectionHelper.mapOf(
+                        // quarkus
+                        "quarkus.infinispan-client.server-list", serverList,
+                        "quarkus.infinispan-client.near-cache-max-entries", "3",
+                        "quarkus.infinispan-client.auth-username", USER,
+                        "quarkus.infinispan-client.auth-password", PASS,
+                        "quarkus.infinispan-client.auth-realm", "default",
+                        "quarkus.infinispan-client.sasl-mechanism", "DIGEST-MD5",
+                        "quarkus.infinispan-client.auth-server-name", "infinispan",
+                        // camel
+                        "camel.component.infinispan.autowired-enabled", "false",
+                        "camel.component.infinispan.hosts", serverList,
+                        "camel.component.infinispan.username", USER,
+                        "camel.component.infinispan.password", PASS,
+                        "camel.component.infinispan.secure", "true",
+                        "camel.component.infinispan.security-realm", "default",
+                        "camel.component.infinispan.sasl-mechanism", "DIGEST-MD5",
+                        "camel.component.infinispan.security-server-name",
+                        "infinispan");
+            } else {
+                return CollectionHelper.mapOf(
+                        // quarkus
+                        "quarkus.infinispan-client.server-list", serverList,
+                        "quarkus.infinispan-client.near-cache-max-entries", "3",
+                        "quarkus.infinispan-client.auth-username", USER,
+                        "quarkus.infinispan-client.auth-password", PASS,
+                        "quarkus.infinispan-client.auth-realm", "default",
+                        "quarkus.infinispan-client.sasl-mechanism", "DIGEST-MD5",
+                        "quarkus.infinispan-client.auth-server-name", "infinispan",
+                        "quarkus.infinispan-client.client-intelligence", "BASIC",
+                        // camel
+                        "camel.component.infinispan.autowired-enabled", "false",
+                        "camel.component.infinispan.hosts", serverList,
+                        "camel.component.infinispan.username", USER,
+                        "camel.component.infinispan.password", PASS,
+                        "camel.component.infinispan.secure", "true",
+                        "camel.component.infinispan.security-realm", "default",
+                        "camel.component.infinispan.sasl-mechanism", "DIGEST-MD5",
+                        "camel.component.infinispan.security-server-name",
+                        "infinispan");
+            }
```

Comment:
Could we perhaps reduce the number of lines a bit? Maybe something like the following?

```suggestion
            Map<String, String> result = CollectionHelper.mapOf(
                        // quarkus
                        "quarkus.infinispan-client.server-list", serverList,
                        "quarkus.infinispan-client.near-cache-max-entries", "3",
                        "quarkus.infinispan-client.auth-username", USER,
                        "quarkus.infinispan-client.auth-password", PASS,
                        "quarkus.infinispan-client.auth-realm", "default",
                        "quarkus.infinispan-client.sasl-mechanism", "DIGEST-MD5",
                        "quarkus.infinispan-client.auth-server-name", "infinispan",
                        // camel
                        "camel.component.infinispan.autowired-enabled", "false",
                        "camel.component.infinispan.hosts", serverList,
                        "camel.component.infinispan.username", USER,
                        "camel.component.infinispan.password", PASS,
                        "camel.component.infinispan.secure", "true",
                        "camel.component.infinispan.security-realm", "default",
                        "camel.component.infinispan.sasl-mechanism", "DIGEST-MD5",
                        "camel.component.infinispan.security-server-name",
                        "infinispan");
            if (SystemUtils.IS_OS_MAC) {
                /* Fix for https://github.com/apache/camel-quarkus/issues/2840 */
                result.put("quarkus.infinispan-client.client-intelligence", "BASIC");
            }
            return result;
```

Reply:
Hi @ppalaga ,

Thanks for the feedback and I will go this way.
I think we need to add this property for windows machine as well. The root cause is same for windows and mac. The Docker desktop can't route request to internal docker IP directly on mac and  windows.

Please see this discussion
https://stackoverflow.com/questions/63465221/running-infinispan-in-docker-on-windows-10
Freeman

Reply:
Thanks, sounds good.